server/public/plugin/client.go

```diff
@@ -26,14 +25,69 @@ func ClientMain(pluginImplementation any) {
 	impl.SetAPI(nil)
 	impl.SetDriver(nil)
 
-	pluginMap := map[string]plugin.Plugin{
+	return map[string]plugin.Plugin{
 		"hooks": &hooksPlugin{hooks: pluginImplementation},
 	}
+}
+
+// WithTestContext provides a context typically used to terminate a plugin from a unit test.
+func WithTestContext(ctx context.Context) func(*plugin.ServeConfig) error {
+	return func(config *plugin.ServeConfig) error {
+		if config.Test == nil {
+			config.Test = &plugin.ServeTestConfig{}
+		}
+
+		config.Test.Context = ctx
+
+		return nil
+	}
+}
+
+// WithTestReattachConfigCh configures the channel to receive the ReattachConfig used to reattach
+// an externally launched plugin instance with the Mattermost server.
+func WithTestReattachConfigCh(reattachConfigCh chan<- *plugin.ReattachConfig) func(*plugin.ServeConfig) error {
+	return func(config *plugin.ServeConfig) error {
+		if config.Test == nil {
+			config.Test = &plugin.ServeTestConfig{}
+		}
+
+		config.Test.ReattachConfigCh = reattachConfigCh
+
+		return nil
+	}
+}
+
+// WithTestCloseCh provides a channel that signals when the plugin exits.
+func WithTestCloseCh(closeCh chan<- struct{}) func(*plugin.ServeConfig) error {
+	return func(config *plugin.ServeConfig) error {
+		if config.Test == nil {
+			config.Test = &plugin.ServeTestConfig{}
+		}
 
-	plugin.Serve(&plugin.ServeConfig{
+		config.Test.CloseCh = closeCh
+
+		return nil
+	}
+}
+
+// Starts the serving of a Mattermost plugin over net/rpc. gRPC is not yet supported.
```

Comment:
Let's not be too hopeful.
```suggestion
// Starts the serving of a Mattermost plugin over net/rpc. gRPC is yet supported.
```

Reply:
Thanks, @hanzei -- I'm not sure what the suggestion here is. This is the same text as before, and `gRPC` is indeed not yet supported. Can you clarify?

Reply:
Woops, messed up the diff. My suggestion is to remove the word `yet`.